src/_articles/language/emulating-functions.md

```diff
@@ -45,14 +47,23 @@ suitably defined object.
 
 ### How does it work?
 
-When x(a<sub>1</sub>, .., a<sub>n</sub>) is evaluated, if it is a normal
-function, it gets called in the normal way. If it isn't we just invoke call()
-on it. If x supports a call() method with suitable arguments it gets called.
-
-Otherwise, `noSuchMethod()` gets invoked. The default implementation of
-noSuchMethod() checks to see whether it was invoked due to an attempt to use call(),
-and if so issues a helpful error message suggesting you might have wanted to use
-a closure.
+When <code>x(a<sub>1</sub>, .., a<sub>n</sub>)</code> is evaluated,
+if it is a normal
+function, it gets called in the normal way. If it isn't we just invoke `call()`
+on it. If `x` supports a `call()` method with suitable arguments it gets called.
+Otherwise, `noSuchMethod()` gets invoked. 
```

Comment:
Amazing, this is true!

We actually just invoke `call` when the static type of `x` is `dynamic`, or it is a class that has a method named `call`, but the other cases are compile-time errors. 

There are other situations where an instance that has a `call` method won't work so smoothly as a function, but I'm really happy to conclude that it still admits such a nice description when we talk about invocations.

So: No problems here whatsoever! ;-)